python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -172,9 +159,9 @@ async def _process_chat_stream_response(
         self,
         response: AsyncStream,
         chat_history: ChatHistory,
-        kernel: "Kernel",
-        tool_call_behavior: Optional[ToolCallBehavior] = None,
-    ) -> AsyncIterable[List[OpenAIStreamingChatMessageContent]]:
+        tool_call_behavior: ToolCallBehavior,
+        kernel: Optional["Kernel"] = None,
+    ) -> AsyncIterable[Tuple[List[OpenAIStreamingChatMessageContent], Optional[FinishReason]]]:
```

Comment:
Just a question: now that this method returns a tuple like a struct, not an iteration of the contents, should this return type still be `AsyncIterable`?

Reply:
but the tuple is yielded, so still asynciterable!